### PR TITLE
External API ready event

### PR DESCRIPTION
**Pull request by saghul**
Please read the commit messages.

### CODE REVIEW

**Comment by saghul** [comment]
Jenkins please test this please.

File: modules/API/external/external_api.js [hunk 1]
```diff
@@ -271,10 +272,10 @@ function parseArguments(args) {
 function parseSizeParam(value) {
     let parsedValue;
 
-    // This regex parses values of the form 100px, 100em, 100pt or 100%.
+    // This regex parses values of the form 100px, 100em, 100pt, 100vh, 100vw or 100%.
     // Values like 100 or 100px are handled outside of the regex, and
     // invalid values will be ignored and the minimum will be used.
-    const re = /([0-9]*\.?[0-9]+)(em|pt|px|%)$/;
+    const re = /([0-9]*\.?[0-9]+)(em|pt|px|vh|vw|%)$/;
```
Comment (horymury) [inline modules/API/external/external_api.js:278]: think we now also have dvh/dvw. maybe add dvh/dvw, lvh/lvw, svh/svw for completion

**Review by horymury — CHANGES_REQUESTED** [review]

File: modules/API/external/external_api.js (hunk 1 above)
Comment (saghul) [reply modules/API/external/external_api.js:278]: TIL, fixed!

**Review by horymury — DISMISSED** [review]

**Comment by saghul** [comment]
Sounds good, I can do that!

File: modules/API/external/external_api.js [hunk 2]
```diff
@@ -271,10 +272,10 @@ function parseArguments(args) {
 function parseSizeParam(value) {
     let parsedValue;
 
-    // This regex parses values of the form 100px, 100em, 100pt or 100%.
+    // This regex parses values of the form 100px, 100em, 100pt, 100vh, 100vw or 100%.
     // Values like 100 or 100px are handled outside of the regex, and
     // invalid values will be ignored and the minimum will be used.
-    const re = /([0-9]*\.?[0-9]+)(em|pt|px|%)$/;
+    const re = /([0-9]*\.?[0-9]+)(em|pt|px|(d|l|s)?vh|(d|l|s)?vw|%)$/;
```
Comment (keremoge) [inline modules/API/external/external_api.js:278]: `((d|l|s)?v)(h|w)` would be shorter than `(d|l|s)?vh|(d|l|s)?vw`

**Comment by saghul** [comment]
> I like this!
> 
> Just an idea: We can call the passed onload function when we receive the ready event. This way we won't need to change the API and depricate the onload function. This way even if the people are using the onload param it will behave correctly. IMHO having the parameter in the constructor has its value because in theory to attach the event you need the external api instance which would have already created the iframe. So theoretically there's a room for a race condition.
> 
> WDYT?

Updated, can you PTAL?

**Review by hristoterezov — APPROVED** [review]

**Pull request merged**